packages/core/data-transfer/lib/providers/local-file-destination-provider.ts

```diff
@@ -90,14 +100,46 @@ class LocalFileDestinationProvider implements IDestinationProvider {
     fs.mkdirSync(path.join(rootDir, 'configuration'));
   }
 
-  rollback(): void | Promise<void> {
+  async close(): Promise<void> {
+    await this.#writeMetadata();
+  }
+
+  rollback(): void {
     fs.rmSync(this.options.file.path, { force: true, recursive: true });
   }
 
   getMetadata() {
     return null;
   }
 
+  async #writeMetadata(): Promise<void> {
+    const metadata = this.#providersMetadata.source;
+
+    if (metadata) {
+      await new Promise((resolve) => {
+        const outStream = this.#getMetadataStream();
+        const data = JSON.stringify(metadata, null, 2);
+
+        Readable.from(data).pipe(outStream).on('close', resolve);
+      });
+    }
+  }
+
+  #getMetadataStream() {
+    const metadataPath = path.join(this.options.file.path, 'metadata.json');
+
+    // Transform streams
+    const transforms: Writable[] = this.#getDataTransformers({ jsonl: false });
+
+    // FS write stream
+    const fileStream = fs.createWriteStream(metadataPath);
```

Comment:
I think I commented on this (or something similar) before. For now this works, but for the archive (and moving encrypt/compress to the same level) we'll have to replace all of the fs writestreams in this file and instead just have one that they all pass through.

Reply:
Yes, I'm currently working on this regarding the archive export, this is temporary & will be updated in the up coming PR